plugins/wdc/wdc-nvme.c

```diff
@@ -1091,28 +1187,45 @@ static bool get_dev_mgment_cbs_data(int fd, __u8 log_id, void **cbs_data)
 		}
 	}
 
-	ret = nvme_get_log(fd, 0xFFFFFFFF, WDC_NVME_GET_DEV_MGMNT_LOG_PAGE_OPCODE,
-			   false, le32_to_cpu(hdr_ptr->length), data);
-	/* parse the data until the List of log page ID's is found */
+	/* get the log page data */
+	ret = nvme_get_log_from_uuid(fd, 0xFFFFFFFF, WDC_NVME_GET_DEV_MGMNT_LOG_PAGE_OPCODE,
+			   false, uuid_ix, le32_to_cpu(hdr_ptr->length), data);
+
 	if (ret) {
 		fprintf(stderr, "ERROR : WDC : Unable to read C2 Log Page data, ret = 0x%x\n", ret);
 		goto end;
 	}
 
+	/* Check the log data to see if the WD version of log page ID's is found */
+
 	length = sizeof(struct wdc_c2_log_page_header);
 	hdr_ptr = (struct wdc_c2_log_page_header *)data;
+	sph = (struct wdc_c2_log_subpage_header *)(data + length);
+	found = wdc_get_dev_mng_log_entry(hdr_ptr->length, log_id, hdr_ptr, &sph);
 
-	while (length < le32_to_cpu(hdr_ptr->length)) {
-		sph = (struct wdc_c2_log_subpage_header *)(data + length);
+	if (found) {
+		*cbs_data = (void *)&sph->data;
+	} else {
 
-		if (le32_to_cpu(sph->entry_id) == log_id) {
+		/* not found with uuid = 1 try with uuid = 0 */
+		if (uuid_ix)
```

Comment:
I don't follow this logic. uuid_ix was initialized to 1, and it there's nothing preceeding this that modified it. When is this 'if' not taken?

Reply:
True... will remove the if